### PR TITLE
rust: use system libs

### PR DESCRIPTION
**Pull request by haampie**

### CODE REVIEW

File: var/spack/repos/builtin/packages/rust/package.py [hunk 1]
```diff
@@ -94,6 +96,9 @@ def setup_dependent_package(self, module, dependent_spec):
         module.cargo = Executable(os.path.join(self.spec.prefix.bin, "cargo"))
 
     def setup_build_environment(self, env):
+        # libssh2-sys
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/rust/package.py]:
```suggestion
        # Manually instruct Cargo dependency libssh2-sys to build with
        # the Spack installed libssh2 package. For more info see
        # https://github.com/alexcrichton/ssh2-rs/issues/173
```

**Comment by alecbcs** [comment]
One quick suggestion to match the comment style throughout the package and give a bit more info. Otherwise looks great! Thank you @haampie for fixing this.

File: var/spack/repos/builtin/packages/rust/package.py (hunk 1 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/rust/package.py]: `libssh2-sys` is a crate of its own (just happens to be in the ssh2-rs repo)

Comment (alecbcs) [reply var/spack/repos/builtin/packages/rust/package.py]: Oop yep you're right, I thought it might have been renamed. Updated in the comment.

**Review by alecbcs — APPROVED** [review]
Looks good to me. Thanks @haampie

**Pull request merged**